compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -478,6 +479,44 @@ fn check_opaque_meets_bounds<'tcx>(
     let _ = infcx.inner.borrow_mut().opaque_type_storage.take_opaque_types();
 }
 
+fn is_enum_of_nonnullable_ptr<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    adt_def: AdtDef<'tcx>,
+    substs: SubstsRef<'tcx>,
+) -> bool {
+    if adt_def.repr().inhibit_enum_layout_opt() {
+        return false;
+    }
+
+    if let Some(ty) = match &adt_def.variants().raw[..] {
+        [var_one, var_two] => match (&var_one.fields[..], &var_two.fields[..]) {
+            ([], [field]) | ([field], []) => Some(field.ty(tcx, substs)),
+            _ => None,
+        },
+        _ => None,
+    } {
+        match ty.kind() {
+            ty::FnPtr(..) | ty::Ref(_, _, _) => true,
+            _ => false,
+        }
+    } else {
+        false
+    }
+}
+
+fn check_static_linkage<'tcx>(tcx: TyCtxt<'tcx>, def_id: LocalDefId) {
+    let ty = tcx.type_of(def_id);
+    if tcx.codegen_fn_attrs(def_id).import_linkage.is_some() {
```

Comment:
Can you check `import_linkage` first, since it is relatively uncommon, and query the type of static later.

Reply:
Done